src/common/hooks/use-transaction-state.ts

```diff
@@ -2,17 +2,30 @@ import { useSelector } from 'react-redux';
 import { RootState } from '@store';
 import {
   selectTransaction,
+  selectTransactionLastTxId,
   selectTransactionLoading,
   selectTransactionError,
 } from '@store/transactions';
 
-export const useTransactionState = (tx_id: string) => {
-  const { transaction, loading, error } = useSelector((state: RootState) => ({
-    transaction: selectTransaction(tx_id)(state),
+export const useTransactionState = (specificTx?: string) => {
+  let transaction = undefined;
+  const { tx_id, loading, error } = useSelector((state: RootState) => ({
+    tx_id: selectTransactionLastTxId(state),
     loading: selectTransactionLoading(state),
     error: selectTransactionError(state),
   }));
+
+  if (specificTx) {
+    transaction = useSelector((state: RootState) => ({
+      transaction: selectTransaction(specificTx)(state),
+    })).transaction;
+  } else if (tx_id) {
+    transaction = useSelector((state: RootState) => ({
+      transaction: selectTransaction(tx_id)(state),
+    })).transaction;
+  }
```

Comment:
What's the difference between a `tx_id` and a `specificTx`?

Reply:
You can use this hook to fetch the most recent TX id searched for, or supply another txid.

Reply:
By fetch you mean, get from store, not make a request?

I just find it a little confusing to read, and afaict, breaks SRP. If we want to get the most recently searched for tx, couldn't we keep this hook as it was before,:
- Persist an array of queried txs
- Where needed, ` const id = selectMostRecentTxId(state); const tx = useTxState(id)`

Reply:
yeah, sure that makes sense! will update